regression/goto-cc-cbmc-shared-options/object-bits/test.c

```diff
@@ -0,0 +1,30 @@
+#include <assert.h>
+#include <stdlib.h>
+
+extern __CPROVER_size_t __CPROVER_malloc_size;
+
+size_t
+find_first_set(const size_t max_malloc_size, const size_t bits_accumulator)
+{
+  // Implemented using recursion to avoid excessive unwinding or specifying an
+  // unwind limit.
+  if(max_malloc_size & 1)
+    return bits_accumulator;
+  return find_first_set(max_malloc_size >> 1, bits_accumulator + 1);
+}
+
+size_t calculate_object_bits()
```

Comment:
IMO it would be unnecessary to reverse engineer the object bits here or even use the value of `__CPROVER_max_malloc_size`. It's probably sufficient to just run `cbmc` with a non-default object bits setting after `goto-cc` and check that it suceeds.

Reply:
The reason I chose to do it this way is that the previous incorrect behaviour resulted in a warning, but might not actually cause a failure without giving an input program which depends on the implications of the `object-bits` configuration. I could have used a negative regex to check that the warning message does not appear in the output. However this would not be a robust test because if the text of the message were to be updated then the corresponding test could easily be missed. This may be a little over-engineered but I am happy that this is a robust test and it is clear what the positive regexes are actually checking for.

Reply:
Yes, that makes sense.